File: Selenium/Activities/Activity22.java
package activities;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.interactions.Actions;

public class Activity22 {
public static void main(String[] args){
    WebDriverManager.firefoxdriver().setup();
    WebDriver driver = new FirefoxDriver();
    driver.get("https://v1.training-support.net/selenium/popups");
    System.out.println("Title is :" + driver.getTitle());
    WebElement hover = driver.findElement(By.className("orange"));
    Actions tooltip = new Actions(driver);
    tooltip.moveToElement(hover).build().perform();
    System.out.println("Sign in Button tooltip is :" + hover.getAttribute("data-tooltip"));
    hover.click();
    driver.findElement(By.id("username")).sendKeys("admin");
    driver.findElement(By.id("password")).sendKeys("password");
    driver.findElement(By.xpath("//button[text() = 'Log in']")).click();
    String confirm = driver.findElement(By.id("action-confirmation")).getText();
    System.out.println("Confirmation Message is :" + confirm);
    driver.close();


}
}

File: Selenium/Activities/Activity22.py
from selenium import webdriver
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions

service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service = service) as driver:
    driver.get("https://v1.training-support.net/selenium/popups")
    print("Tile is :" ,driver.title)
    hover = driver.find_element(By.CLASS_NAME,'orange')
    action = ActionChains(driver)
    action.move_to_element(hover).perform()
    print("Sign in tooltip is :" ,hover.get_attribute("data-tooltip"))
    wait =WebDriverWait(driver,10)
    hover.click()
    wait.until(expected_conditions.visibility_of_element_located((By.ID,'username')))
    driver.find_element(By.ID,'username').send_keys('admin')
    driver.find_element(By.ID,'password').send_keys('password')
    driver.find_element(By.XPATH,"//button[text() = 'Log in']").click()
    confirm = driver.find_element(By.ID,'action-confirmation').text
    print("Confirmation Message is :" + confirm)
    driver.close()
